src/main/java/at/hannibal2/skyhanni/features/minion/InfernoMinionFeatures.kt

```diff
@@ -0,0 +1,75 @@
+package at.hannibal2.skyhanni.features.minion
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.jsonobjects.repo.InfernoMinionFuelsJson
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.events.InventoryCloseEvent
+import at.hannibal2.skyhanni.events.InventoryFullyOpenedEvent
+import at.hannibal2.skyhanni.events.LorenzToolTipEvent
+import at.hannibal2.skyhanni.events.RepositoryReloadEvent
+import at.hannibal2.skyhanni.utils.ItemUtils.name
+import at.hannibal2.skyhanni.utils.KeyboardManager
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NEUInternalName
+import at.hannibal2.skyhanni.utils.StringUtils.matches
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class InfernoMinionFeatures {
+    private val config get() = SkyHanniMod.feature.minions
+    private val infernoMinionTitlePattern by RepoPattern.pattern(
+        "minion.infernominiontitle",
+        "Inferno Minion .*"
+    )
+    private var fuelItemIds = listOf<NEUInternalName>()
+    private var isInventory = false
+
+    @SubscribeEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        val data = event.getConstant<InfernoMinionFuelsJson>("InfernoMinionFuels")
+        fuelItemIds = data.minionFuels
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        isInventory = infernoMinionTitlePattern.matches(event.inventoryName)
+    }
+
+    @SubscribeEvent
+    fun onInventoryClose(event: InventoryCloseEvent) {
+        isInventory = false
+    }
+
+    @SubscribeEvent
+    fun onSlotClick(event: GuiContainerEvent.SlotClickEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.infernoFuelBlocker) return
+        if (!isInventory) return
+
+        val containsFuel = NEUInternalName.fromItemNameOrNull(event.container.getSlot(19).stack.name.toString()) in fuelItemIds
+        if (!containsFuel) return
+
+        if (event.slot?.slotNumber == 19 || event.slot?.slotNumber == 53) {
+            if (KeyboardManager.isModifierKeyDown()) return
+            event.cancel()
+        }
+    }
+
+    @SubscribeEvent
+    fun onTooltip(event: LorenzToolTipEvent) {
+        if (!config.infernoFuelBlocker) return
+        if (!isInventory) return
+
+        val containsFuel = NEUInternalName.fromItemNameOrNull(event.itemStack.name.toString()) in fuelItemIds
+        if (!containsFuel) return
+
+        if (event.slot.slotNumber == 19) {
+            event.toolTip.add("")
+            event.toolTip.add("§c§l[SkyHanni] is blocking you from taking this out! (Hold CTRL to override)")
```

Comment:
Please split this up in two or three lines so that the item tooltip does not get too wide.
Maybe hide the tooltip while the bypass keybind is pressed.
![image](https://github.com/hannibal002/SkyHanni/assets/24389977/722b1c59-850b-49ea-9b36-e5ef4a5878be)

Reply:
Maybe format it similar to this.
And also give it the gray background. [See in HideNotClickableItems](https://github.com/hannibal002/SkyHanni/blob/93af7607b3d5c1061bc845f097fe8be94c5e2043/src/main/java/at/hannibal2/skyhanni/features/inventory/HideNotClickableItems.kt#L107)
![image](https://github.com/hannibal002/SkyHanni/assets/24389977/4d1bd790-045a-419b-bfa3-994df708c538)